elastic/kibana
# [Index Management] Fix bug in mappings fields

Fixes https://github.com/elastic/kibana/issues/156202

## Summary

This PR fixes the bug in Mappings fields from the Component template wizard by changing the Subtype parameter so that it handles the case when the value in the Subtype dropdown menu is cleared, similarly to [what is done in the Type parameter](https://github.com/elastic/kibana/blob/801919702ae03d8946c4a5677d387566f43f615c/x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/field_parameters/type_parameter.tsx#L75).

**How to test:**

1. Navigate to Index Management, go to Component templates tab.
2. Start the wizard to create a new component template.
3. Go to the Mappings step and add a field of type **Numeric** or **Range**, and click on the Subtype dropdown menu.
4. Verify that pressing backspace in the Subtype dropdown menu doesn't make the page to crash.
5. After adding the mapping field, click the pencil icon to edit this field and in the flyout click in the Numeric/Range type dropdown.
6. Verify that pressing backspace doesn't make the page to crash.
